tests/test_generics.py

```diff
@@ -0,0 +1,395 @@
+    )
+
+
+def test_get_fields_type_map_for_subscripted() -> None:
+    _T = TypeVar("_T")
+
+    @dataclasses.dataclass(frozen=True, slots=True, kw_only=True)
+    class Xxx(Generic[_T]):
+        xxx: _T
+
+    actual = get_fields_type_map(Xxx[str])
+    assert actual == {"xxx": str}
+
+
+def test_get_fields_class_map() -> None:
+    _T = TypeVar("_T")
+
+    @dataclasses.dataclass()
+    class Base1(Generic[_T]):
+        a: str
+        b: str
+        c: str
+
+    @dataclasses.dataclass()
+    class Base2(Base1[int]):
+        a: str
+        b: str
+        d: str
+        e: str
+
+    @dataclasses.dataclass()
+    class BaseG:
+        f: str
+        g: str
+
+    @dataclasses.dataclass()
+    class Base3(Base2, BaseG):
+        a: str
+        d: str
+        f: str
+        h: str
+
+    actual = get_fields_class_map(Base3)
+    assert actual == {
+        "a": Base3,
+        "b": Base2,
+        "c": Base1,
+        "d": Base3,
+        "e": Base2,
+        "f": Base3,
+        "g": BaseG,
+        "h": Base3,
+    }
+
+
+def test_get_class_type_var_map_with_inheritance() -> None:
+    _T1 = TypeVar("_T1")
+    _T2 = TypeVar("_T2")
+    _T3 = TypeVar("_T3")
+
+    @dataclasses.dataclass()
+    class NonGeneric:
+        pass
+
+    @dataclasses.dataclass()
+    class Aaa(Generic[_T1, _T2]):
+        pass
+
+    @dataclasses.dataclass()
+    class Bbb(Generic[_T1], Aaa[int, _T1]):
+        pass
+
+    @dataclasses.dataclass()
+    class Ccc(Generic[_T3]):
+        pass
+
+    @dataclasses.dataclass()
+    class Ddd(Generic[_T1, _T2, _T3], Bbb[_T2], Ccc[_T1], NonGeneric):
+        pass
+
+    actual = get_class_type_var_map(Ddd[bool, str, float])
+    assert actual == {
+        Aaa: {
+            _T1: int,
+            _T2: str,
+        },
+        Bbb: {
+            _T1: str,
+        },
+        Ccc: {
+            _T3: bool,
+        },
+        Ddd: {
+            _T1: bool,
+            _T2: str,
+            _T3: float,
+        },
+    }
+
+
+def test_get_class_type_var_map_with_incompatible_inheritance() -> None:
+    _T = TypeVar("_T")
+
+    @dataclasses.dataclass()
+    class Aaa(Generic[_T]):
+        pass
+
+    @dataclasses.dataclass()
+    class Bbb(Aaa[int]):
+        pass
+
+    @dataclasses.dataclass()
+    class Ccc(Bbb, Aaa[str]):  # type: ignore
+        pass
+
+    with pytest.raises(ValueError, match="Incompatible Base class") as e:
+        get_class_type_var_map(Ccc)
+    assert "<locals>.Aaa'> with generic args {~_T: <class 'int'>} and {~_T: <class 'str'>}" in e.value.args[0]
+
+
+def test_get_class_type_var_map_with_duplicated_generic_inheritance() -> None:
+    _T = TypeVar("_T")
+
+    @dataclasses.dataclass()
+    class NonGeneric:
+        pass
+
+    @dataclasses.dataclass()
+    class Aaa(Generic[_T]):
+        pass
+
+    @dataclasses.dataclass()
+    class Bbb(Aaa[int], NonGeneric):
+        pass
+
+    @dataclasses.dataclass()
+    class Ccc(Bbb, Aaa[int], NonGeneric):
```

Comment:
```suggestion
    class Ccc(Bbb, Aaa[str], NonGeneric):
```

to highlight the issue more explicitly

Reply:
no.

it is correct.

take a look on the assertion.

case with same Generics but different Args in test above
`test_get_class_type_var_map_with_incompatible_inheritance`